client/src/components/Tasks.jsx

```diff
@@ -1,39 +1,43 @@
-import React, { Component } from "react";
+import React from "react";
 import OneInputForm from "./OneInputForm";
 import Task from "./Task";
+import { useListTasks } from '../hooks/useListTasks'
+import { useTasksAndLists } from "../hooks/useTasksAndLists";
 
 function Tasks(props) {
-
 	const listID = props.match.params.id;
-
-	const tasks = props.lists[listID].tasks.map((taskID, index) => {
-		const task = props.tasks[taskID];
-
-		if (!task) return null;
-
-		const { complete, text } = task;
+	// This is another custom hook, that hides logic related to
+	// a certain list in one place.
+	const {
+		title,
+		listTasks,
+		addTaskToList,
+		removeTaskFromList,
+	} = useListTasks(listID);
```

Comment:
Added a second custom hook. Let me know if this makes sense :)

It's not flawless right now. An error occurs when refreshing sometimes, but not sure why.

Reply:
Is this to help improve the DRYness of the code?

Reply:
Essentially, yes. It allows us to "hide" logic related to our state from the components.